arcgis-ios-sdk-samples/Maps/Display a map SwiftUI/SwiftUIMapView.swift

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+struct SwiftUIMapView {
+    let map: AGSMap
+    let graphicsOverlays: [AGSGraphicsOverlay]
+    
+    init(
+        map: AGSMap,
+        graphicsOverlays: [AGSGraphicsOverlay] = []
+    ) {
+        self.map = map
+        self.graphicsOverlays = graphicsOverlays
+    }
+    
+    private var onSingleTapAction: ((CGPoint, AGSPoint) -> Void)?
+}
+
+extension SwiftUIMapView {
+    func onSingleTap(action: @escaping (CGPoint, AGSPoint) -> Void) -> Self {
+        var copy = self
+        copy.onSingleTapAction = action
+        return copy
+    }
+}
+
+extension SwiftUIMapView: UIViewRepresentable {
+    typealias UIViewType = AGSMapView
+    
+    func makeCoordinator() -> Coordinator {
+        Coordinator(
+            onSingleTapAction: onSingleTapAction
+        )
+    }
+    
+    func makeUIView(context: Context) -> AGSMapView {
+        let uiView = AGSMapView()
+        uiView.map = map
+        uiView.graphicsOverlays.setArray(graphicsOverlays)
+        uiView.touchDelegate = context.coordinator
+        return uiView
+    }
+    
+    func updateUIView(_ uiView: AGSMapView, context: Context) {
+        if map != uiView.map {
+            uiView.map = map
+        }
+        if graphicsOverlays != uiView.graphicsOverlays as? [AGSGraphicsOverlay] {
+            uiView.graphicsOverlays.setArray(graphicsOverlays)
+        }
+        context.coordinator.onSingleTapAction = onSingleTapAction
+    }
+}
+
+extension SwiftUIMapView {
+    /// You can use this coordinator to implement common Cocoa patterns, such as
+    /// delegates, data sources, and responding to user events via target-action.
+    class Coordinator: NSObject {
+        var onSingleTapAction: ((CGPoint, AGSPoint) -> Void)?
+        
+        init(
+            onSingleTapAction: ((CGPoint, AGSPoint) -> Void)?
```

Comment:
After reading the SwiftUI tutorials, I'm still not fully understanding what a [`Coordinator`](https://developer.apple.com/documentation/swiftui/uiviewrepresentable/coordinator) should hold. In the [tutorial](https://developer.apple.com/tutorials/swiftui/interfacing-with-uikit#:~:text=declare%20a%20nested%20coordinator%20class%20inside%20pageviewcontroller) it holds its parent, but IMO it can hold anything, including those which can also be the stored properties of the parent view struct. A few possible options...

- Hold its parent (so that it can access its parent's properties)
- Hold the target-action delegate methods and corresponding action closures
- Hold everything. In our case, the map, the overlays array, and the `onSingleTapAction` closure

Can you help me to understand which is preferred and why? Thanks.